huggingface_sb3/push_to_hub.py

```diff
@@ -7,30 +7,309 @@
+    # Step 1: Select the tags
+    model_card = select_tags(env_id)
+
+    # Step 2: Generate the model card
+    model_card += f"""
+  # **{model_name}** Agent playing **{env_id}**
+  This is a trained model of a **{model_name}** agent playing **{env_id}** using the [stable-baselines3 library](https://github.com/DLR-RM/stable-baselines3).
+  ## Evaluation Results
+  """
+
+    model_card += f"""
+  mean_reward={mean_reward:.2f} +/- {std_reward}
+  """
+
+    model_card += """
+  ## Usage (with Stable-baselines3)
+  TODO: Add your code
+  """
+
+    return model_card
+
+
+def _create_model_card(repo_dir: Path, generated_model_card):
+    """Creates a model card for the repository.
+    TODO: Add metrics to model-index
+    TODO: Use information from common model cards
+    :param repo_dir: repository directory
+    :param generated_model_card: model card generated by _generate_model_card() method
     """
     readme_path = repo_dir / "README.md"
     readme = ""
     if readme_path.exists():
-      with readme_path.open("r", encoding="utf8") as f:
-          readme = f.read()
+        with readme_path.open("r", encoding="utf8") as f:
+            readme = f.read()
     else:
-      readme = README_TEMPLATE
+        readme = generated_model_card
     with readme_path.open("w", encoding="utf-8") as f:
         f.write(readme)
 
+
+def package_to_hub(model,
+                   model_name: str,
+                   model_architecture: str,
+                   env_id: str,
+                   eval_env,
+                   repo_id: str,
+                   commit_message: str,
+                   is_deterministic=True,
+                   n_eval_episodes=10,
+                   use_auth_token=True,
+                   local_repo_path="hub",
+                   video_length=1000,
+                   ):
+    """
+      Evaluate, Generate a video and Upload a model to Hugging Face Hub.
+      This method does the complete pipeline:
+      - It evaluates the model
+      - It generates the model card
+      - It generates a replay video of the agent
+      - It pushes everything to the hub
+      This is a work in progress function, if it does not work, use push_to_hub method
+      :param model: trained model
+      :param model_name: name of the model zip file
+      :param model_architecture: name of the architecture of your model (DQN, PPO, A2C, SAC...)
+      :param env_id: name of the environment
+      :param eval_env: environment used to evaluate the agent
+      :param repo_id: repo_id: id of the model repository from the Hugging Face Hub
+      :param commit_message: commit message
+      :param is_deterministic: use deterministic or stochastic actions (by default: True)
+      :param n_eval_episodes: number of evaluation episodes (by default: 10)
+      :param use_auth_token
+      :param local_repo_path: local repository path
+      :param video_length:
+      :param video_length: length of the video (in timesteps)
+      """
+    print(
```

Comment:
Let's use logging instead of printing across the repo

Reply:
:heavy_check_mark: